services/mpa/README.md

```diff
@@ -0,0 +1,192 @@
+# Multi Party Authentication
+
+WARNING: This document describes the intended state. https://github.com/Snowflake-Labs/sansshell/issues/346 tracks implementation.
+
+This module enables [multi-party authorization](https://en.wikipedia.org/wiki/Multi-party_authorization) for any sansshell command. Approval data is stored in-memory in sansshell-server.
+
+## User flow
+
+MPA must be explicitly requested. When requested, the MPA flow will be used regardless of whether a policy would allow a command without MPA.
+
+1. A user issues commands.
+
+   ```bash
+   $ sanssh -mpa -targets=1.2.3.4 -justification emergency exec run /bin/echo hi
+   Waiting for approval for 1-2345-6789. Command for approving:
+       sanssh -targets=1.2.3.4 mpa approve 1-2345-6789
+   ```
+
+2. The approver views the commands and approves it.
+
+   ```bash
+   $ sanssh -targets=1.2.3.4 mpa list
+   1-2345-6789
+   $ sanssh -targets=1.2.3.4 mpa get 1-2345-6789
+   user: firstuser
+   justification: emergency
+   method: /Exec.Exec/Run
+   message: {
+      "command": "/bin/echo",
+      "args": ["hi"]
+   }
+   $ sanssh -targets=1.2.3.4 mpa approve 1-2345-6789
+   ```
+
+3. If the user's command is still running, it will complete. If the user had stopped their command, they can rerun it and the approval will still be valid as long as the command's input remains the same.
```

Comment:
nit: should probably say “as long as the command’s input remains the same and the sansshell-server approval is still in memory (the binary has not restarted, and it has not been cleared by a user)"

Reply:
Good call, updated.